Documentation/devel/architecture.md

```diff
@@ -102,3 +102,60 @@ This will activate all the reaper services when one of the targets is activated,
+
+This is done with the following chain:
+
+```
+ -----------            -----------            ------------
+|           |          |           |          |            |
+|   Fetch   |--------->|   Store   |--------->|   Render   |
+|           |          |           |          |            |
+ -----------            -----------            ------------
+
+```
+
+* Fetch: in the fetch phase rkt will retrieve the required images. The fetching implementation depends on the provided image argument (image string like `example.com/app:v1.0`, hash, https URL, file).
+* Store: in the store phase the fetched images will be saved to the local store. The local store is a cache for fetched images and additional related data.
+* Render: in the render phase, a renderer will pull the required images from the store and render them so they can be easily used as a stage2 content.
+
+
+These three logical blocks are implemented inside rkt in this way:
+
+```
+ ------------            ---------------                -------------                    ------------------------
+|            |          |               |              |             |     overlayfs    |                        |
+|  Fetchers  |--------->|  Image Store  |<-------------|  TreeStore  |<-----------------|  Stage1-2 fs contents  |
+|            |          |               |<----         |             |           -------|                        |
+ ------------            ---------------      \         -------------           /        ------------------------
+                                               \                               /
+                                                \       -----------------     /
+                                                 \     |                 |   /
+                                                  -----| Direct stage1-2 |---
+                                                       |   renderer      |
+                                                       |                 |
+                                                        -----------------
+```
+
+Currently rkt implements the appc spec, in future additional specs like (OCI) may be implemented keeping the same logic but changing the implementations.
+
+* Fetchers: the fetchers, depending on the provided image argument will do image discovery (if an image string is provided) and/or fetching (using the discovered urls or the provided urls). The fetchers may read some information from the store to check if an image is already in the store or to retrieve additional data like http caching information. Once fetched the images will be verified against their signatures and then saved in the Image Store. If an image has some dependencies, they will be discovered and fetched. See also [image fetching behavior](../image-fetching-behavior.md).
```

Comment:
> [dependencies][appc-dependencies]

Since it's not possible to point to markdown line numbers the best I can do is to point to the image manifest schema section https://github.com/appc/spec/blob/master/spec/aci.md#image-manifest-schema

suggestions?